Tests/BuildTests/PrepareForIndexTests.swift

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2024 Apple Inc. All rights reserved.
+//
+// This document is the property of Apple Inc.
+// It is considered confidential and proprietary.
+//
+// This document may not be reproduced or transmitted in any form,
+// in whole or in part, without the express written permission of
+// Apple Inc.
+
+import Build
+import Foundation
+import LLBuildManifest
+@_spi(SwiftPMInternal)
+import SPMTestSupport
+import TSCBasic
+import XCTest
+
+class PrepareForIndexTests: XCTestCase {
+    func testPrepare() throws {
+        let (graph, fs, scope) = try macrosPackageGraph()
+
+        let plan = try BuildPlan(
+            destinationBuildParameters: mockBuildParameters(prepareForIndexing: true),
+            toolsBuildParameters: mockBuildParameters(prepareForIndexing: true),
+            graph: graph,
+            fileSystem: fs,
+            observabilityScope: scope
+        )
+
+        let builder = LLBuildManifestBuilder(plan, fileSystem: fs, observabilityScope: scope)
+        let manifest = try builder.generatePrepareManifest(at: "/manifest")
+
+        // Make sure we're still building swift modules
+        XCTAssertNotNil(manifest.commands["<SwiftSyntax-debug.module>"])
+        // Make sure we're not building things that link
+        XCTAssertNil(manifest.commands["C.Core-debug.exe"])
```

Comment:
Should this be a per platform check?

Reply:
Yup, fix incoming. :)

Reply:
Made sure tests work on Mac and Linux which forced me to be more platform agnostic.